### ISSUE TITLE
Idea

### ISSUE THREAD

**Issue by mastermood** [muhammadmuzzammil1998/BitcoinBot#5]
Is it possible to invoke BitcoinBot at custom timed intervals in chat? I.E. timed commands to do >btc USD say from 1m-4h? Would be a nice feature I think.

**Comment by muhammadmuzzammil1998** [comment]
Yeah that would be a cool feature to have. I will definitely look into that. Thanks!

**Comment by mastermood** [comment]
Also I thought of another cool feature. Text to speech. so when 

/tts $7000
/tts $9000
/tts $8000

with custom timer. It's so simple to do, and helps you get alerts if BTC is doing something bad.

**Comment by muhammadmuzzammil1998** [comment]
Yeah sure...

**Comment by mastermood** [comment]
It's actually not a bad idea because I don't think there's any other bot that does that. I wouldn't make it a default, but an option perhaps. It's like a line or two right? But timed messages that show that at custom intervals is pretty dope. Then getting an alert every X minutes/hours/etc. makes it even robust but also sticks to keeping it simple the way you intended. 

Just a suggestion, but I think you'll get more servers adding it/exposure since no other one does it :D

**Comment by muhammadmuzzammil1998** [comment]
Yes, @mastermood I do think that this is a great idea and I wasn't being sarcastic. If that hurt your feelings, I am sorry. I was kinda busy when I read the notification so I replied a short answer. 

I haven't started working on it because of my exams and rest assured that I will code that `tts` command for BitcoinBot.

What do you think? For normal tts reply, ```>btc USD tts``` is better or ```>btc USD -t``` or something else?

Thank you for supporting the Project.

**Comment by mastermood** [comment]
I'm not sure on what flag works. I thought it usually works as /tts >btc USD and the output hides the /tts flag. 

Btw, I've noticed a slight delay of the price of about 3-5 minutes when it outputs the price. Is that due to API limitations. I hosted the scripts on my own server for the bot, but I got the same delay. 

Oh - one other suggestion: Consider adding option to choose either GDAX or Bitfinex API for pulling data as an option. Why? Because overall market follow Bitfinex and/or GDAX. They are both good, but the Bitfinex prices are what generally impact all the other markets.

**Comment by muhammadmuzzammil1998** [comment]
Well, the delay *is* from coinbase and there is nothing I can do about it. It could be their "cool down" mechanism to prevent spam requests.

GDAX and Bitfinex, while both are good services, they lack a variety of real-currencies options to choose from in getting the price. For instance, bitfinex does not even support INR (Indian Rupee), it reports an error of `{"message":"Unknown symbol"}`.

**Comment by mastermood** [comment]
IMO, the best is USD since most websites do analyses and metrics on that.

**Comment by mastermood** [comment]
Did you see what Nabeel wrote? He said "To remove the delay load the prices at a fixed interval in an async manner in a go routine and use channels to share the value with the main go routine."

**Comment by muhammadmuzzammil1998** [comment]
Yeah, I got the email. I talked to him about this matter. But the thing is, if I do that, it won't be "real-time" and "current" as we all know that bitcoin's rate fluctuate. So, he deleted his comment.

**Comment by mastermood** [comment]
![unknown](https://user-images.githubusercontent.com/9539678/39674496-c8580354-511a-11e8-9c9a-d5b91642b0bc.png)

I think the sidebar is better when the order is like this:

"Playing $9520 | 10 ms | >btc help"

Because the price gets cuts off if it's the middle entry.

**Comment by muhammadmuzzammil1998** [comment]
Please open another issue for this. 

Thanks.
